core/extension/gdextension_interface.h

```diff
@@ -2721,6 +2721,30 @@ typedef void (*GDExtensionInterfacePlaceHolderScriptInstanceUpdate)(GDExtensionS
  */
 typedef GDExtensionScriptInstanceDataPtr (*GDExtensionInterfaceObjectGetScriptInstance)(GDExtensionConstObjectPtr p_object, GDExtensionObjectPtr p_language);
 
+/**
+ * @name object_set_script_instance
+ * @since 4.5
+ *
+ * Set the script instance data attached to this object.
```

Comment:
It would be helpful to explain the difference between `object_set_script_instance` and `object_set_script_and_instance` since they seems to do very similar behavior (see my previous comments ^^)